### PR TITLE
Changing method type in ReorderMethodArguments

### PR DESCRIPTION
**Pull request by joanvr**
## What's changed?
This recipe was not changing the method type, so even when changing the order of the arguments, it was matching again the next cycle and "reordering" the arguments again. If we update the methodType of the methodInvocation it's not needed to "hack" the tests forcing one cycle only, and thus we can check the idempotency of the recipe.

## What's your motivation?
I was using this recipe in another recipe, and not getting the expected results, because the recipe was applied two times (one in each cycle).

## Anyone you would like to review specifically?
@knutwannheden @timtebeek 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've used the IntelliJ auto-formatter on affected files

### CODE REVIEW

File: rewrite-java-test/src/test/java/org/openrewrite/java/ReorderMethodArgumentsTest.java [hunk 1]
```diff
@@ -31,17 +27,7 @@ void reorderArguments() {
...
-              new String[]{"n", "m", "s"}, null, null, null),
-            toRecipe(() -> new JavaVisitor<>() {
-                @Override
-                public J visitLiteral(J.Literal literal, ExecutionContext p) {
-                    if (literal.getType() == JavaType.Primitive.String) {
-                        doAfterVisit(new ChangeLiteral<>(literal, l -> "anotherstring"));
-                    }
-                    return super.visitLiteral(literal, p);
-                }
-            })
```
Comment (joanvr) [inline rewrite-java-test/src/test/java/org/openrewrite/java/ReorderMethodArgumentsTest.java:43]: I'm not sure what was the purpose of this? It seems not related to the actual test of the recipe, so I decided to remove it

File: rewrite-java-test/src/test/java/org/openrewrite/java/ReorderMethodArgumentsTest.java [hunk 2]
```diff
@@ -85,9 +71,8 @@ public void test() {
     @Test
     void reorderArgumentsWithNoSourceAttachment() {
         rewriteRun(
-          spec -> spec.recipe(new ReorderMethodArguments("a.A foo(..)",
-              new String[]{"s", "n"}, new String[]{"n", "s"}, null, null))
-            .cycles(1).expectedCyclesThatMakeChanges(1),
+          spec -> spec.recipe(new ReorderMethodArguments("a.A foo(String,..)",
```
Comment (joanvr) [inline rewrite-java-test/src/test/java/org/openrewrite/java/ReorderMethodArgumentsTest.java:74]: had to make the matcher a bit more specific, otherwise it was always matching no matter if we changed the method Type

File: rewrite-java-test/src/test/java/org/openrewrite/java/ReorderMethodArgumentsTest.java [hunk 3]
```diff
@@ -123,15 +108,14 @@ public void test() {
     @Test
     void reorderArgumentsWhereOneOfTheOriginalArgumentsIsVararg() {
         rewriteRun(
-          spec -> spec.recipe(new ReorderMethodArguments("a.A foo(..)",
-              new String[]{"s", "o", "n"}, null, null, null))
-            .cycles(1).expectedCyclesThatMakeChanges(1),
+          spec -> spec.recipe(new ReorderMethodArguments("a.A foo(String,Integer,..)",
```
Comment (joanvr) [inline rewrite-java-test/src/test/java/org/openrewrite/java/ReorderMethodArgumentsTest.java:111]: had to make the matcher a bit more specific, otherwise it was always matching no matter if we changed the method Type

File: rewrite-java-test/src/test/java/org/openrewrite/java/ReorderMethodArgumentsTest.java [hunk 4]
```diff
@@ -123,15 +108,14 @@ public void test() {
...
+            new String[]{"s", "o", "n"}, null, null, null)),
           java(
             """
               package a;
               public class A {
                  public void foo(String s, Integer n, Object... o) {}
-                 public void bar(String s, Object... o) {}
+                 public void foo(String s, Object... o) {}
```
Comment (joanvr) [inline rewrite-java-test/src/test/java/org/openrewrite/java/ReorderMethodArgumentsTest.java:118]: changed this to foo, otherwise we were changing to a non existing method, and this bar was useless.

File: rewrite-java/src/main/java/org/openrewrite/java/ReorderMethodArguments.java [hunk 5]
```diff
@@ -144,12 +146,19 @@ public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, Execu
                         // this is a varargs argument
                         List<JRightPadded<Expression>> varargs = originalArgs.subList(fromPos, originalArgs.size());
                         reordered.addAll(varargs);
+                        for (int j = 0; j < varargs.size(); j++) {
+                            reorderedNames.add(name + j);
```
Comment (joanvr) [inline rewrite-java/src/main/java/org/openrewrite/java/ReorderMethodArguments.java:150]: Not sure what to do here, since this is the varargs argument, and can have N arguments, I've decided to put in the names the name of the original vararg + index.

**Review by knutwannheden — APPROVED** [review]
LGTM. Is there any issue we should link it to?

**Comment by timtebeek** [comment]
> LGTM. Is there any issue we should link it to?

This was discovered in
- https://github.com/openrewrite/rewrite-okhttp/pull/5
